docs/more_details/integration_tests.md

```diff
@@ -0,0 +1,121 @@
+
+This _client_ object is provided by Gosoline, whenever at least one of a test suite's methods has the above signature.
+
+[embedmd]:# (../../examples/getting_started/integration/api_test.go /func \(s \*ApiTestSuite\) SetupApiDefinitions/ /return definer.ApiDefiner\n}/)
+```go
+func (s *ApiTestSuite) SetupApiDefinitions() apiserver.Definer {
+	return definer.ApiDefiner
+}
+```
+
+Implementing _SetupApiDefinitions_ when configuring your test will inform Gosoline that an API is being tested, thus the _client_ object will contain meaningful data.
+
+Forgeting to implement _SetupApiDefinitions_, and the trying to run an _resty.Client_ test case, will result in an error reminding you that `the suite has to implement the TestingSuiteApiDefinitionsAware interface`.
+
+### ApiServer integration tests
+
+The main advantage of the _client_ object is it allows you to control when requests are executed and gives you access to the endpoint, by providing the host & port. Therefore, if you, for example, want to time your requests, this is the way to go.  If you only want to make a standard API call and read the response, you can use Gosoline's _ApiServerTestCase_. These type of tests issue HTTP calls and compare their responses against a predefined value, but they need to follow a predefined structure. Below we can see how the above _resty client_ test would look like as an _ApiServerTestCase_:
+
+[embedmd]:# (../../examples/getting_started/integration/api_test.go /func \(s \*ApiTestSuite\) Test_Euro/ /\n}/)
+```go
+func (s *ApiTestSuite) Test_Euro() *suite.ApiServerTestCase {
+	return &suite.ApiServerTestCase{
+		Method:             http.MethodGet,
+		Url:                "/euro/10/GBP",
+		Headers:            map[string]string{},
+		ExpectedStatusCode: http.StatusOK,
+		Assert: func(response *resty.Response) error {
+			result, err := strconv.ParseFloat(string(response.Body()), 64)
+			s.NoError(err)
+			s.Equal(8.0, result)
+
+			return nil
+		},
+	}
+}
+```
+
+Notice the predefined structure of an _ApiServerTestCase_: it is an object that has fields for all the information needed in performing an HTTP call, the expected status code, and a method that tests the result for correctness.
+
+In the example from `examples/getting_started/integration/api_test.go` we see two types of test cases (_resty.Client_ and _ApiServerTestCase_) in the same test suite. In fact, all three types of tests can be part of the same test suite.
+
+### Stream integration tests
+
+If you have a Gosoline application that takes its input from a stream, you need a test which can run your application locally, send data to a stream, and check your application's output for correctness. 
+
+The stream-consumer application, found in more_details/stream-consumer, reads unsigned integers from the inputEvent input, increments them by one, and publishes them to the publisher-outputEvent output. Below is an extract from its integration test:
```

Comment:
As you said later on, the input is actually called `consumer`. The *queue* is called `inputEvent` (leaving out the prefixes), but that queue name doesn't play a role during the integration test (only the input name)